SteamAPI/GenericInventory.cs

```diff
@@ -132,33 +141,41 @@ public Inventory GetInventory(int appId, ulong contextId)
             catch
             {
                 throw new GenericInventoryException();
-            }            
+            }
         }
 
-        public void AddForeignInventory(SteamID steamId, int appId, ulong contextId)
+        public void AddForeignInventory(SteamID steamId, int appId, ulong contextId, bool onlyTradeableItems)
         {
-            var inventory = FetchForeignInventory(steamId, appId, contextId);
+            var inventory = FetchForeignInventory(steamId, appId, contextId, onlyTradeableItems);
             if (!_inventories.HasAppId(appId))
                 _inventories[appId] = new BotInventories.ContextInventory();
             if (inventory != null && !_inventories[appId].HasContextId(contextId))
                 _inventories[appId].Add(contextId, inventory);
         }
 
-        private Inventory FetchForeignInventory(SteamID steamId, int appId, ulong contextId)
+        private Inventory FetchForeignInventory(SteamID steamId, int appId, ulong contextId, bool onlyTradeableItems, int count = 5000)
         {
-            var inventoryUrl = string.Format("http://steamcommunity.com/trade/{0}/foreigninventory/?sessionid={1}&steamid={2}&appid={3}&contextid={4}", steamId.ConvertToUInt64(), _steamWeb.SessionId, steamId.ConvertToUInt64(), appId, contextId);
-            return FetchInventory(inventoryUrl, steamId, appId, contextId);
+            return FetchInventory(steamId, appId, contextId, count, onlyTradeableItems);
         }
 
-        private Inventory FetchInventory(string inventoryUrl, SteamID steamId, int appId, ulong contextId, int start = 0)
+        private Inventory FetchInventory(SteamID steamId, int appId, ulong contextId, int count = 5000, bool onlyTradeableItems = true, string lastAssetId = "")
         {
-            inventoryUrl = inventoryUrl + "&start=" + start;
+            var inventoryUrl = string.Format("http://steamcommunity.com/inventory/{0}/{1}/{2}?count={3}", steamId.ConvertToUInt64(), appId, contextId, count);
+
+            if (onlyTradeableItems)
+                inventoryUrl += "?trading=1";
```

Comment:
This would probably break the url. Also I'm not sure if `trading=1` actually does anything with the new endpoint.